src/mechanics/cell/SimpleAssember.h

```diff
@@ -0,0 +1,144 @@
+    {
+        GlobalDofMatrixSparse hessian = ProperlyResizedGlobalMatrix(dofTypes);
+
+        for (NuTo::CellInterface* cell : cells)
+        {
+            const DofVector<int> numbering = cell->DofNumbering();
+            const DofMatrix<double> cellHessian = cell->Hessian0();
+
+            for (const DofType* dofI : dofTypes)
+            {
+                for (const DofType* dofJ : dofTypes)
+                {
+                    const Eigen::VectorXi& numberingDofI = numbering[*dofI];
+                    const Eigen::VectorXi& numberingDofJ = numbering[*dofJ];
+                    const Eigen::MatrixXd& cellHessianDof = cellHessian(*dofI, *dofJ);
+
+                    const int numActiveDofsI = mNumActiveDofs[*dofI];
+                    const int numActiveDofsJ = mNumActiveDofs[*dofJ];
+
+                    for (int i = 0; i < numberingDofI.rows(); ++i)
+                    {
+                        for (int j = 0; j < numberingDofJ.rows(); ++j)
+                        {
+                            const int globalDofNumberI = numberingDofI[i];
+                            const int globalDofNumberJ = numberingDofJ[j];
+                            const double globalDofValue = cellHessianDof(i, j);
+
+                            const bool activeI = globalDofNumberI < numActiveDofsI;
+                            const bool activeJ = globalDofNumberJ < numActiveDofsJ;
+
+                            if (activeI)
+                            {
+                                if (activeJ)
+                                {
+                                    hessian.JJ(*dofI, *dofJ).coeffRef(globalDofNumberI, globalDofNumberJ) +=
+                                            globalDofValue;
+                                }
+                                else
+                                {
+                                    hessian.JK(*dofI, *dofJ)
+                                            .coeffRef(globalDofNumberI, globalDofNumberJ - numActiveDofsJ) +=
+                                            globalDofValue;
+                                }
+                            }
+                            else
+                            {
+                                if (activeJ)
+                                {
+                                    hessian.KJ(*dofI, *dofJ)
+                                            .coeffRef(globalDofNumberI - numActiveDofsI, globalDofNumberJ) +=
+                                            globalDofValue;
+                                }
+                                else
+                                {
+                                    hessian.KK(*dofI, *dofJ)
+                                            .coeffRef(globalDofNumberI - numActiveDofsI,
+                                                      globalDofNumberJ - numActiveDofsJ) += globalDofValue;
+                                }
+                            } // argh. any better ideas?
```

Comment:
- Is there not a function that takes lists of "i", "j" and "value"?
- Would it be possible/advantageous to assemble everything into KK first, and then "reorder" in a second step?

Reply:
Yeah, you mean an `Eigen::Triplet<>`. You see the usage [here](https://eigen.tuxfamily.org/dox/group__SparseQuickRefPage.html). This sould be more efficient than the `coeffref` thing. IMO we have to perform benchmarks to see, what is best.